lib/open_project/authentication/strategies/warden/session.rb

```diff
@@ -0,0 +1,31 @@
+module OpenProject
+  module Authentication
+    module Strategies
+      module Warden
+        ##
+        # Temporary strategy necessary as long as the OpenProject authentication has
+        # not been unified in terms of Warden strategies and is only locally
+        # applied to the API v3.
+        class Session < ::Warden::Strategies::Base
+          def valid?
+            session
+          end
+
+          def authenticate!
+            user = user_id ? User.find(user_id) : User.anonymous
+
+            success! user
+          end
+
+          def user_id
+            Hash(session)['user_id']
```

Comment:
I assume the trick here is, that this is not the place in the session where warden stores its information?

Meaning that this strategy gets the session information in the "old" way, while the warden store is a new way of doing it?